analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,8 +125,11 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
-        ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
+    private bool FileIsIncludedInAnalysis(SonarLintXmlReader sonarLintXml, string filePath) =>
```

Comment:
The `File` doesn't seem to be needed given the arguments
```suggestion
    private bool IsIncludedInAnalysis(SonarLintXmlReader sonarLintXml, string filePath) =>
```